test/unit/option_sources/id_params_test.rb

```diff
@@ -0,0 +1,30 @@
+require File.join(File.dirname(__FILE__), '../test_helper')
+
+describe HammerCLIForeman::OptionSources::IdParams do
+  describe "get_options" do
+    class IdParamsTestCommand < HammerCLIForeman::CreateCommand
+      resource :hostgroups
+      build_options
+    end
+
+    let(:hg_cmd) { IdParamsTestCommand.new("", { :adapter => :csv, :interactive => false }) }
+    let(:id_params_source) { HammerCLIForeman::OptionSources::IdParams.new(hg_cmd) }
+
+    it "skips param when set" do
+      hg_cmd.expects(:get_resource_id).with { |res| res.name != :domains }.at_least(0).returns(nil)
```

Comment:
Do we actually need `.at_least(0)`? See my other comment.

Reply:
My thinking was that `at_least(0)` stands for `any times` which equivalent I didn't found in expectations docs. The thing is that the resource I use for testing in the command has other id params. When I set the expectation for the one param I care about I also need to cover the other calls to :get_resource_ids otherwise I get unexpected invocations. I used `at_least(0)` to indicate I don't care about the number of invocations. Is there better way to do this?

The other case has just one other param so the default `once()` works here. I've added at_least(0) to be consistent.